application/comit_node/src/swap_protocols/mod.rs

```diff
@@ -15,8 +16,27 @@ pub use self::{
     timestamp::Timestamp,
 };
 
-#[derive(Debug)]
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Hash, Copy)]
+pub enum HashFunction {
+    #[serde(rename = "sha-256")]
+    SHA256,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Hash)]
 pub enum SwapProtocol {
-    Rfc003,
+    Rfc003(HashFunction),
     Unknown(String),
 }
+
+#[cfg(test)]
```

Comment:
:mega: I don't think you need this test, because you'd just be testing `serde`'s already tested derived implementation of `Serialize`.

Reply:
This test verifies that the serde renaming works as expected.  Perhaps unnecessary for more experienced Rust hackers but I think for me it is useful to leave it in, especially since from the comment above we are about to change the case of the serialized string `SHA-256`.

Reply:
@tcharding I absolutely see and understand your point as I did similarly when starting with Rust last year. However, here what you are really stating is that the serde `rename` feature works (and that "SHA-256" is spelled correctly on line 21). I don't think we should test serde in our codebase.

Reply:
Okay, explained like this I get it.  Thanks, will remove this test.